shell/src/main/resources/interpreter-setting.json

```diff
@@ -38,6 +38,13 @@
         "defaultValue": "",
         "description": "Kerberos principal",
         "type": "string"
+      },
+      "zeppelin.shell.interpolation": {
+        "envName": null,
+        "propertyName": null,
```

Comment:
doesn't this need a propertyName?

Reply:
Since this parameter is interpreter-level, not global, the _envName_ and _propertyName_ are not needed. I think the _envName_ is for values to be parsed from the file _zeppelin-env.sh_, and the _propertyName_ is for values to be parsed from the file _zeppelin-site.xml_

But request @zjffdu to confirm my assumptions above.

Reply:
envName yes, but I think it should have propertyName? it's at least consistent to all other properties.
I'll defer to @zjffdu though

Reply:
envName is not necessary. It is not used by zeppelin-env.sh. Actually it is never used, I planned to remove it from interpreter-setting.json.  BTW the structure of `interpreter-setting.json` (property of each interpreter) and `interpreter.json` (all the interpreter saved configuration) is not consistent, we may need to redesign the structure to make them consistent.

Reply:
@felixcheung the _propertyName_ field in both _interpreter-setting.json_ files have been restored. Thanks for the catch.

@zjffdu thanks for the clarification.

I read the code and documents for the interpreter configuration area, and believe both need a closer look and many improvements are possible. If you are planning an effort for this after 0.8.0, I would love to help. My strengths are mainly Scala and Java, and I have also written professionally before.